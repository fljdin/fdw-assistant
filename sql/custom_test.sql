CREATE SCHEMA source;

create table source.clients (
  id bigint not null,
  firstname varchar(255),
  lastname varchar(255) not null,
  created_at timestamp(0),
  updated_at timestamp(0),
  is_active smallint,
  is_imported smallint,
  source varchar(255),
  updated_by bigint,
  imported_at timestamp(0),
  deleted_at timestamp(0)
);

COMMENT ON COLUMN source.clients.is_active IS 'bool(%s)';

COMMENT ON COLUMN source.clients.is_imported IS 'bool(%s)';

create table public.clients (
  id bigint not null,
  firstname varchar(255),
  lastname varchar(255) not null,
  created_at timestamp(0),
  updated_at timestamp(0),
  imported_at timestamp(0),
  deleted_at timestamp(0),
  updated_by bigint,
  is_active boolean,
  is_imported boolean,
  source varchar(255)
);

create table source.documents (request text);

COMMENT ON COLUMN source.documents.request IS '%s::json';

create table public.documents (request json);

INSERT INTO config (source, target, pkey, priority, parts, trunc, condition, batchsize) VALUES
  ('source.clients', 'public.clients', null, 1, 1, true, null, null),
  ('source.documents', 'public.documents', null, 2, 1, true, null, null);

SELECT * FROM config;

SELECT invocation FROM plan() \gexec
