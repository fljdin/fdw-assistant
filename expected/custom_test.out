INSERT INTO config (source, target, pkey, priority, parts, trunc, condition, batchsize) VALUES
  ('source.clients', 'public.clients', null, 1, 1, true, null, null),
  ('source.documents', 'public.documents', null, 2, 1, true, null, null),
  ('source."MixedCaseTable"', 'public."MixedCaseTable"', null, 3, 1, true, null, null);
INSERT 0 3
SELECT * FROM config;
         source          |         target          | pkey | priority | parts | trunc | condition | batchsize 
-------------------------+-------------------------+------+----------+-------+-------+-----------+-----------
 source.clients          | public.clients          |      |        1 |     1 | t     |           |          
 source.documents        | public.documents        |      |        2 |     1 | t     |           |          
 source."MixedCaseTable" | public."MixedCaseTable" |      |        3 |     1 | t     |           |          
(3 rows)

SELECT invocation FROM plan() \gexec
CALL assistant.copy(1);
NOTICE:  Executing: TRUNCATE public.clients
NOTICE:  Executing: SELECT count(1) FROM source.clients 
NOTICE:  Executing: INSERT INTO public.clients SELECT id, firstname, lastname, created_at, updated_at, imported_at, REPLACE(deleted_at, '0000-00-00', '1970-01-01')::timestamp, updated_by, bool(is_active), bool(is_imported), source FROM source.clients   
CALL
CALL assistant.copy(2);
NOTICE:  Executing: TRUNCATE public.documents
NOTICE:  Executing: SELECT count(1) FROM source.documents 
NOTICE:  Executing: INSERT INTO public.documents SELECT request::json FROM source.documents   
CALL
CALL assistant.copy(3);
NOTICE:  Executing: TRUNCATE public."MixedCaseTable"
NOTICE:  Executing: SELECT count(1) FROM source."MixedCaseTable" 
NOTICE:  Executing: INSERT INTO public."MixedCaseTable" SELECT "MixedCaseColumn" FROM source."MixedCaseTable"   
CALL
